presto-hive/src/main/java/io/prestosql/plugin/hive/recordwriters/SequenceFileFormatRecordWriter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.recordwriters;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.Utilities;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.SequenceFile.Writer;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.Reporter;
+
+import java.io.IOException;
+
+public class SequenceFileFormatRecordWriter
+        implements ExtendedRecordWriter
+{
+    private long finalWrittenBytes = -1;
+    private final Writer sequenceFileWriter;
+    private static final BytesWritable EMPTY_KEY = new BytesWritable();
+
+    public SequenceFileFormatRecordWriter(Path path, JobConf jobConf, Class value, boolean isCompressed) throws IOException
+    {
+        sequenceFileWriter = Utilities.createSequenceWriter(jobConf, path.getFileSystem(jobConf), path, BytesWritable.class, value, isCompressed, Reporter.NULL);
+    }
+
+    @Override
+    public long getWrittenBytes()
+    {
+        if (finalWrittenBytes != -1) {
+            return finalWrittenBytes;
+        }
+        try {
+            return sequenceFileWriter.getLength();
+        }
+        catch (IOException e) {
+            return 0; // do nothing
+        }
+    }
+
+    @Override
+    public void write(Writable writable) throws IOException
+    {
+        sequenceFileWriter.append(EMPTY_KEY, writable);
+    }
+
+    @Override
+    public void close(boolean b) throws IOException
+    {
+        if (finalWrittenBytes == -1) {
```

Comment:
Can we change this to close first, then call `getLength()`? Or is it not valid to ask for position after close?

Reply:
asking for position after close throwing is invalid - since close set the internal output stream to `null`. I added flush because the local file output stream (used in tests) keeps data in memory and flushes after a certain number of written bytes - leading to failure of tests.
On that note, I think that parquet would also face this problem since the data would be written only when a row group is flushed.

Reply:
Nice find. We might need to do the same for Parquet. (though eventually it will be replaced with the native writer)

Reply:
Yes, will look for a fix for it. From first looks, it seems like `InternalParquetRecordWriter#getDataSize` can be used - will patch that as a part of this change.